src/tools/illink/test/Mono.Linker.Tests.Cases/DataFlow/AttributePropertyDataflow.cs

```diff
@@ -12,24 +12,148 @@ namespace Mono.Linker.Tests.Cases.DataFlow
 	[ExpectedNoWarnings]
 	class AttributePropertyDataflow
 	{
-		[KeptAttributeAttribute (typeof (KeepsPublicConstructorsAttribute))]
-		[KeptAttributeAttribute (typeof (KeepsPublicMethodsAttribute))]
-		[KeptAttributeAttribute (typeof (KeepsPublicFieldsAttribute))]
-		[KeptAttributeAttribute (typeof (TypeArrayAttribute))]
-		[KeepsPublicConstructors (Type = typeof (ClassWithKeptPublicConstructor))]
-		[KeepsPublicMethods (TypeName = "Mono.Linker.Tests.Cases.DataFlow.AttributePropertyDataflow+ClassWithKeptPublicMethods")]
-		[KeepsPublicFields (Type = null, TypeName = null)]
-		[TypeArray (Types = new Type[] { typeof (AttributePropertyDataflow) })]
-		// Trimmer only for now - https://github.com/dotnet/linker/issues/2273
-		[ExpectedWarning ("IL2026", "--ClassWithKeptPublicMethods--", ProducedBy = Tool.Trimmer | Tool.NativeAot)]
 		public static void Main ()
 		{
-			typeof (AttributePropertyDataflow).GetMethod (nameof (Main)).GetCustomAttribute (typeof (KeepsPublicConstructorsAttribute));
-			typeof (AttributePropertyDataflow).GetMethod (nameof (Main)).GetCustomAttribute (typeof (KeepsPublicMethodsAttribute));
+			AttributesOnMethod.Test ();
+			AttributesOnProperty.Test ();
+			AttributesOnField.Test ();
+			AttributesOnEvent.Test ();
+			typeof (AttributePropertyDataflow).GetMethod ("Main").GetCustomAttribute (typeof (KeepsPublicConstructorsAttribute));
+			typeof (AttributePropertyDataflow).GetMethod ("Main").GetCustomAttribute (typeof (KeepsPublicMethodsAttribute));
 			RecursivePropertyDataFlow.Test ();
+			RecursiveMethodDataFlow.Test ();
 			RecursiveEventDataFlow.Test ();
 			RecursiveFieldDataFlow.Test ();
-			RecursiveMethodDataFlow.Test ();
+		}
+
+		class AttributesOnMethod
+		{
+			[Kept]
+			[KeptAttributeAttribute (typeof (KeepsPublicConstructorsAttribute))]
+			[KeptAttributeAttribute (typeof (KeepsPublicMethodsAttribute))]
+			[KeptAttributeAttribute (typeof (KeepsPublicFieldsAttribute))]
+			[KeptAttributeAttribute (typeof (TypeArrayAttribute))]
+			[KeepsPublicConstructors (Type = typeof (ClassWithKeptPublicConstructor))]
+			[KeepsPublicMethods (TypeName = "Mono.Linker.Tests.Cases.DataFlow.AttributePropertyDataflow+AttributesOnMethod+ClassWithKeptPublicMethods")]
+			[KeepsPublicFields (Type = null, TypeName = null)]
+			[TypeArray (Types = new Type[] { typeof (AttributePropertyDataflow) })]
+			// Trimmer only for now - https://github.com/dotnet/linker/issues/2273
+			[ExpectedWarning ("IL2026", "--ClassWithKeptPublicMethods--", ProducedBy = Tool.Trimmer | Tool.NativeAot)]
+			public static void Test () {
```

Comment:
For this to work in NativeAOT the `Test` method must be marked as reflectable (otherwise we drop custom attributes if I remember correctly).